src/compiler/utilities.ts

```diff
@@ -2317,11 +2317,14 @@ namespace ts {
     }
 
     function getSourceOfAssignment(node: Node): Node | undefined {
-        return isExpressionStatement(node) &&
-            node.expression && isBinaryExpression(node.expression) &&
-            node.expression.operatorToken.kind === SyntaxKind.EqualsToken
-            ? node.expression.right
-            : undefined;
+        if(isExpressionStatement(node) && isBinaryExpression(node.expression) && node.expression.operatorToken.kind === SyntaxKind.EqualsToken) {
+            let cur = node.expression;
+            while (isBinaryExpression(cur.right) && cur.right.operatorToken.kind === SyntaxKind.EqualsToken) {
+                cur = cur.right;

```

Comment:
From a quick glance, I think this is a re-implementation of `getRightMostAssignedExpression`?

Reply:
Looks like it! I found `getInitializerOfBinaryExpression`, but not getRightMostAssignedExpression.